zrml/global-disputes/src/benchmarks.rs

```diff
@@ -281,7 +283,10 @@ benchmarks! {
             }
             .into(),
         );
-        assert!(T::Currency::free_balance(&reward_account) == 0u128.saturated_into());
+        assert_eq!(
+            T::Currency::free_balance(&reward_account),
+            reward_before.checked_rem(&o.into()).unwrap(),
+        );
```

Comment:
Applied this commits on branch `upgrade_0929` and started verification. When executing the tests, it works. When executing the benchmark with `steps > 1`, the benchmark fails. It is valid for `steps = 1`, because `n mod 1 = 0 , n ∈ ℤ`, but panics when `steps > 1`. Strangely the tests seem to set the balance of the reward account to `prev_balance mod o`.

```sh
./target/release/zeitgeist benchmark pallet --chain=dev --steps=10 --repeat=1000 --pallet=zrml_global_disputes --extrinsic=reward_outcome_owner_with_funds --execution=Native --heap-pages=4096 --template=./misc/weight_template.hbs --output=./dev/null

====================

Version: 0.3.7-ffef740d554

   0: sp_panic_handler::set::{{closure}}
   1: <alloc::boxed::Box<F,A> as core::ops::function::Fn<Args>>::call
             at /rustc/4a14677239bbd9a3892d165ee24513c1a05820e4/library/alloc/src/boxed.rs:1954:9
      std::panicking::rust_panic_with_hook
             at /rustc/4a14677239bbd9a3892d165ee24513c1a05820e4/library/std/src/panicking.rs:673:13
   2: std::panicking::begin_panic_handler::{{closure}}
             at /rustc/4a14677239bbd9a3892d165ee24513c1a05820e4/library/std/src/panicking.rs:560:13
   3: std::sys_common::backtrace::__rust_end_short_backtrace
             at /rustc/4a14677239bbd9a3892d165ee24513c1a05820e4/library/std/src/sys_common/backtrace.rs:138:18
   4: rust_begin_unwind
             at /rustc/4a14677239bbd9a3892d165ee24513c1a05820e4/library/std/src/panicking.rs:556:5
   5: core::panicking::panic_fmt
             at /rustc/4a14677239bbd9a3892d165ee24513c1a05820e4/library/core/src/panicking.rs:142:14
   6: core::panicking::assert_failed_inner
   7: core::panicking::assert_failed
   8: core::ops::function::FnOnce::call_once{{vtable.shim}}
   9: zrml_global_disputes::benchmarks::<impl frame_benchmarking::utils::Benchmarking for zrml_global_disputes::pallet::Pallet<T>>::run_benchmark
  10: <battery_station_runtime::Runtime as frame_benchmarking::utils::runtime_decl_for_Benchmark::Benchmark<sp_runtime::generic::block::Block<sp_runtime::generic::header::Header<u64,sp_runtime::traits::BlakeTwo256>,sp_runtime::generic::unchecked_extrinsic::UncheckedExtrinsic<sp_runtime::multiaddress::MultiAddress<<<sp_runtime::MultiSignature as sp_runtime::traits::Verify>::Signer as sp_runtime::traits::IdentifyAccount>::AccountId,()>,battery_station_runtime::Call,sp_runtime::MultiSignature,(frame_system::extensions::check_non_zero_sender::CheckNonZeroSender<battery_station_runtime::Runtime>,frame_system::extensions::check_spec_version::CheckSpecVersion<battery_station_runtime::Runtime>,frame_system::extensions::check_tx_version::CheckTxVersion<battery_station_runtime::Runtime>,frame_system::extensions::check_genesis::CheckGenesis<battery_station_runtime::Runtime>,frame_system::extensions::check_mortality::CheckMortality<battery_station_runtime::Runtime>,frame_system::extensions::check_nonce::CheckNonce<battery_station_runtime::Runtime>,frame_system::extensions::check_weight::CheckWeight<battery_station_runtime::Runtime>,pallet_transaction_payment::ChargeTransactionPayment<battery_station_runtime::Runtime>)>>>>::dispatch_benchmark
  11: battery_station_runtime::api::dispatch
  12: std::thread::local::LocalKey<T>::with
  13: sc_executor::native_executor::WasmExecutor<H>::with_instance::{{closure}}
  14: sc_executor::wasm_runtime::RuntimeCache::with_instance
  15: <sc_executor::native_executor::NativeElseWasmExecutor<D> as sp_core::traits::CodeExecutor>::call
  16: sp_state_machine::execution::StateMachine<B,H,Exec>::execute_aux
  17: sp_state_machine::execution::StateMachine<B,H,Exec>::execute
  18: frame_benchmarking_cli::pallet::command::<impl frame_benchmarking_cli::pallet::PalletCmd>::run
  19: sc_cli::runner::Runner<C>::sync_run
  20: zeitgeist::command::run
  21: zeitgeist::main
  22: std::sys_common::backtrace::__rust_begin_short_backtrace
  23: std::rt::lang_start::{{closure}}
  24: core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &F>::call_once
             at /rustc/4a14677239bbd9a3892d165ee24513c1a05820e4/library/core/src/ops/function.rs:286:13
      std::panicking::try::do_call
             at /rustc/4a14677239bbd9a3892d165ee24513c1a05820e4/library/std/src/panicking.rs:464:40
      std::panicking::try
             at /rustc/4a14677239bbd9a3892d165ee24513c1a05820e4/library/std/src/panicking.rs:428:19
      std::panic::catch_unwind
             at /rustc/4a14677239bbd9a3892d165ee24513c1a05820e4/library/std/src/panic.rs:137:14
      std::rt::lang_start_internal::{{closure}}
             at /rustc/4a14677239bbd9a3892d165ee24513c1a05820e4/library/std/src/rt.rs:148:48
      std::panicking::try::do_call
             at /rustc/4a14677239bbd9a3892d165ee24513c1a05820e4/library/std/src/panicking.rs:464:40
      std::panicking::try
             at /rustc/4a14677239bbd9a3892d165ee24513c1a05820e4/library/std/src/panicking.rs:428:19
      std::panic::catch_unwind
             at /rustc/4a14677239bbd9a3892d165ee24513c1a05820e4/library/std/src/panic.rs:137:14
      std::rt::lang_start_internal
             at /rustc/4a14677239bbd9a3892d165ee24513c1a05820e4/library/std/src/rt.rs:148:20
  25: main
  26: __libc_start_call_main
             at ./csu/../sysdeps/nptl/libc_start_call_main.h:58:16
  27: __libc_start_main_impl
             at ./csu/../csu/libc-start.c:392:3
  28: _start


Thread 'main' panicked at 'assertion failed: `(left == right)`
  left: `0`,
 right: `2`', /home/sea212/Documents/github/zeitgeist/zrml/global-disputes/src/benchmarks.rs:286

This is a bug. Please report it at:

	https://github.com/zeitgeistpm/zeitgeist/issues

Error: Input("Error executing and verifying runtime benchmark: Runtime panicked: assertion failed: `(left == right)`\n  left: `0`,\n right: `2`")
```

We can do a workaround by supplying to different `assertions` based on whether we are in a `test` environment or not. However, it would be good to know the cause of it.

```suggestion
        #[cfg(not(test))]
        assert_eq!(
            T::Currency::free_balance(&reward_account),
            0u128.saturated_into(),
        );
        #[cfg(test)]
        assert_eq!(
            T::Currency::free_balance(&reward_account),
            reward_before.checked_rem(&o.into()).unwrap(),
        );
```

Reply:
> Strangely the tests seem to set the balance of the reward account to `prev_balance mod o`.

That's expected behavior. We're splitting the previous balance between `o` people, so everybody gets `prev_balance // o` and the remainder (literally) is `prev_balance mod o`.